src/main.rs

```diff
@@ -0,0 +1,547 @@
+//! STAR Randomness web service
```

Comment:
some ideas for splits:
- request handlers funcs and axum stuff in main.rs
- error stuff in error.rs
- epoch_update_loop in update.rs

Reply:
I split the source as follows:

- `main.rs` just the config and axum app
- `handler.rs` request handlers and associated types, including the Error enum
- `update.rs` for the epoch update background task
- `tests.rs` for the test code

The `handler::Error` enum and related code are just there to propagate `?` to the axum response machinery, so I think it makes sense to keep it with the handlers rather than in a separate file. It has to be `pub` so the router can access them, but main.rs doesn't need to make direct reference to it, and the variants aren't shared with any other components of the app.

Reply:
the split looks good to me. in the future, if we ever need to use the error enum outside of our handlers, we should extract it into error.rs